### DISCUSSION TITLE
Admin password hashes generated by Ansible are not working: how to create a proper hash using Ansible??

### DISCUSSION THREAD

**Discussion by johanneskastl** [wg-easy/wg-easy#1516] [Q&A] [answered]
Dear all,

I am playing around with a demo setup using vagrant and Ansible.

Almost everything works like a charm, the only thing is that I cannot get wg-easy to accept the bcrypt password hashed created by Ansible.

Using the password hash given in [the docs](https://github.com/wg-easy/wg-easy/blob/master/How_to_generate_an_bcrypt_hash.md) works.

Using one generated by Ansible does not. The main different seems to be that wgpw uses 12 rounds, while Ansible complains that it needs to be at least 1000 rounds. A password with 1000 rounds does not work with the wg-easy UI.

And wgpw does not include a salt in the password.

```
# working
$2b$12$coPqCsPtcFO.Ab99xylBNOW4.Iu7OOA2/ZIboHN6/oyxca3MWo7fW
# not working
$6$rounds=656000$1234567890abcde$yeo01g32/VB/HOvx4qJyGvbZGIEw6Tpmz2ox3bAEVwXccqGtDfy2tTs22Z63Jtxcxbe4/4FpN17KQJRJKswWt.
 ```

The password is generated by `'vagrant-libvirt' | password_hash(ident='bcrypt', salt='1234567890abcde')`

# Question

1. Is there any documentation on what hash that is exactly? `bcrypt` seems to be ambiguous...
2. Does anyone have an idea how to create such a hash using Ansible? Running wgpw from Ansible is hacky...

Thanks in advance!

Kind Regards,
Johannes

**Comment by kaaax0815** [comment, marked as answer]
bcrypt is a standard. the hash you provided is most certainly not valid bcrypt
`$6$` means its a sha512 hash, thats why its not working

bcrypt has a salt. specifying it manually is bad btw (should be random)

Tried running it locally
After installing `passlib` and using `password_hash('bcrypt')` it generated a correct hash

**Reply by johanneskastl** [reply]
Damn, you are right. OK, bitten by Ansible once again. Sorry for the noise!



(I am aware of the security implications by specifying the salt. For the demo setup it is fine, and using a salt makes this task idempotent, aka the password task is not shown as changed every time Ansible runs, just because the salt changes...)
